### PR TITLE
Add CloneSet definition

### PR DESCRIPTION
**Pull request by FillZpp**
### Ⅰ. Describe what this PR does
Add CloneSet definition

### CODE REVIEW

File: pkg/apis/apps/v1alpha1/cloneset_types.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	// recreating Pod when possible. Currently, only image update of pod spec is allowed. Any other changes to the pod
+	// spec will fall back to ReCreate CloneSetUpdateStrategyType where pod will be recreated.
+	InPlaceIfPossibleCloneSetUpdateStrategyType CloneSetUpdateStrategyType = "InPlaceIfPossible"
+	// InPlaceOnlyCloneSetUpdateStrategyType indicates that we will in-place update Pod instead of
+	// recreating pod. Currently we only allow image update for pod spec. Any other changes to the pod spec will be
+	// rejected by kube-apiserver
+	InPlaceOnlyCloneSetUpdateStrategyType CloneSetUpdateStrategyType = "InPlaceOnly"
+)
+
+type CloneSetInPlaceUpdateStrategy struct {
```
[the remark this answers is not in the document]
Comment (FillZpp) [reply pkg/apis/apps/v1alpha1/cloneset_types.go]: Reserved structure. Advanced features for in-place update like `ImagePreDownload` will be added into it later.

File: pkg/apis/apps/v1alpha1/cloneset_types.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	// selector is a label query over pods that should match the replica count.
+	// It must match the pod template's labels.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+	Selector *metav1.LabelSelector `json:"selector"`
+
+	// template describes the pods that will be created.
+	Template v1.PodTemplateSpec `json:"template"`
+
+	// volumeClaimTemplates is a list of claims that pods are allowed to reference.
+	// Note that PVC will be deleted when its pod has been deletes.
```
[the remark this answers is not in the document]
Comment (FillZpp) [reply pkg/apis/apps/v1alpha1/cloneset_types.go]: done

File: pkg/apis/apps/v1alpha1/cloneset_types.go [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+	// ReadyReplicas is the number of Pods created by the CloneSet controller that have a Ready Condition.
+	ReadyReplicas int32 `json:"readyReplicas"`
+
+	// UpdatedReplicas is the number of Pods created by the CloneSet controller from the CloneSet version
+	// indicated by updateRevision.
+	UpdatedReplicas int32 `json:"updatedReplicas"`
+
+	// UpdatedReadyReplicas is the number of Pods created by the CloneSet controller from the CloneSet version
+	// indicated by updateRevision and have a Ready Condition.
+	UpdatedReadyReplicas int32 `json:"updatedReadReplicas"`
```
Comment (Fei-Guo) [inline pkg/apis/apps/v1alpha1/cloneset_types.go]: I think there is a typo here, which causes the API violation notice in api/api-rules/violation_exceptions.list, should be  `json:"updatedReadyReplicas"`

Comment (FillZpp) [reply pkg/apis/apps/v1alpha1/cloneset_types.go]: done

File: api/openapi-spec/swagger.json [hunk 4]
```diff
@@ -131,6 +131,248 @@
...
+        "replicas": {
+          "description": "Replicas is the number of Pods created by the CloneSet controller.",
+          "type": "integer",
+          "format": "int32"
+        },
+        "updateRevision": {
+          "description": "UpdateRevision, if not empty, indicates the latest revision of the CloneSet.",
+          "type": "string"
+        },
+        "updatedReadReplicas": {
```
Comment (Fei-Guo) [inline api/openapi-spec/swagger.json]: This is not updated，did you run `make`?

Comment (FillZpp) [reply api/openapi-spec/swagger.json]: done

**Pull request merged**